test/Microsoft.ApplicationInsights.AspNetCore.Tests/JavaScript/ApplicationInsightsJavaScriptTest.cs

```diff
@@ -10,27 +15,93 @@ public static class ApplicationInsightsJavaScriptTest
         public static void SnippetWillBeEmptyWhenInstrumentationKeyIsNotDefined()
         {
             var telemetryConfigurationWithNullKey = new TelemetryConfiguration();
-            var snippet = new JavaScriptSnippet(telemetryConfigurationWithNullKey);
-            Assert.Equal(string.Empty, snippet.FullScript.ToString());
+            var snippet = new JavaScriptSnippet(telemetryConfigurationWithNullKey, GetOptions(false), null);
+            Assert.Equal(string.Empty, snippet.FullScript);
         }
 
         [Fact]
         public static void SnippetWillBeEmptyWhenInstrumentationKeyIsEmpty()
         {
-            var telemetryConfigurationWithEmptyKey = new TelemetryConfiguration { InstrumentationKey = string.Empty };
-            var snippet = new JavaScriptSnippet(telemetryConfigurationWithEmptyKey);
-            Assert.Equal(string.Empty, snippet.FullScript.ToString());
+            var telemetryConfigurationWithEmptyKey = new TelemetryConfiguration {InstrumentationKey = string.Empty};
+            var snippet = new JavaScriptSnippet(telemetryConfigurationWithEmptyKey, GetOptions(false), null);
+            Assert.Equal(string.Empty, snippet.FullScript);
         }
 
+        [Fact]
+        public static void SnippetWillBeEmptyWhenTelemetryDisabled()
+        {
+            var telemetryConfigurationWithEmptyKey = new TelemetryConfiguration
+            {
+                InstrumentationKey = "NonEmpty",
+                DisableTelemetry = true
+            };
+            var snippet = new JavaScriptSnippet(telemetryConfigurationWithEmptyKey, GetOptions(false), null);
+            Assert.Equal(string.Empty, snippet.FullScript);
+        }
 
         [Fact]
         public static void SnippetWillIncludeInstrumentationKeyAsSubstring()
         {
             string unittestkey = "unittestkey";
+            var telemetryConfiguration = new TelemetryConfiguration {InstrumentationKey = unittestkey};
+            var snippet = new JavaScriptSnippet(telemetryConfiguration, GetOptions(false), null);
+            Assert.Contains("instrumentationKey: '" + unittestkey + "'", snippet.FullScript);
+        }
+
+        [Fact]
+        public static void SnippetWillIncludeAuthUserNameIfEnabledAndAuthenticated()
+        {
+            string unittestkey = "unittestkey";
             var telemetryConfiguration = new TelemetryConfiguration { InstrumentationKey = unittestkey };
-            var snippet = new JavaScriptSnippet(telemetryConfiguration);
-            Assert.Contains("instrumentationKey: '" + unittestkey + "'", snippet.FullScript.ToString());
+            var snippet = new JavaScriptSnippet(telemetryConfiguration, GetOptions(true), GetHttpContextAccessor("username", true));
+            Assert.Contains("setAuthenticatedUserContext(\"username\")", snippet.FullScript);
+        }
+
+        [Fact]
+        public static void SnippetWillNotIncludeAuthUserNameIfEnabledAndAuthenticated()
```

Comment:
Other than the "\\" in the username in the next test what is the difference between this test and the one after it?
Why does "\\" specify whether or not someone is authenticated?  Couldn't there be an identity object, with a name which does not contain slashes, that is for an authenticated user from some system other than Active Directory?